sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_intel_matrix.asciidoc

```diff
@@ -21,135 +19,380 @@
+  wi_element &operator/=(const T &rhs);
+
+  std::tuple<size_t, size_t> get_coord();
+};
+
+} // namespace sycl::ext::intel::experimental::matrix
+```
+
+In the following example `wi_data_c` is a reference to the WI owned
+portion of the joint matrix `matC`. As such `wi_data_c[i] OP rhs`
+updates the corresponding matrix element in the joint_matrix `matC`.
+Vectorization along the sub group dimension will get enabled
+automatically to vectorize the contiguous portion of the matrix.
+
+
+```c++
+auto wi_data_c = get_wi_data(sg, matC);
+for (int i = 0; i < wi_data_c.length(); i++)
+  wi_data_c[i] *= alpha;    // Note that the indexing here "i"
+  //is in the vector owned by a WI, not in the matrix C
 ```
 
+===== Work-item data to joint matrix mapping coordinates
+The `wi_data` and `wi_element` classes provide access to the matrix
+elements that are local to the calling work-item. However, the
+distribution of matrix elements to each work-item is
+implementation-defined, so application code cannot assume any fixed
+distribution. Instead, application code can use the `get_coord` method
+to query the matrix coordinates of an individual `wi_element`.
+
+`get_coord` returns [row,col] coordinates of the current object
+`wi_element` of the joint matrix.  The code above results into the following:
 
-## VNNI/Packed Layout
-Intel AMX and Intel XMX compute assumes that the B tile register (src1) is in the VNNI format as they need 32bit of K-data in A and B to be contiguous in memory. 
-The VNNI blocking factor is 2 in the case of 16-bit types, and it is 4 in the case of 8-bit types. While the current implementation assumes that the matrix has been already packed by the user for performance reasons, the layout information is needed to inform the implementation about this transformation.  The following example illustrates how a matrix in `row_major` layout is transformed into the `packed` layout for a 16-bit type.
+```c++
+auto data = get_wi_data(sg, tA);
+// each WI calculates local sum of rows
+for (int i = 0; i < data.length(); ++i) {
+  auto [row, col] = data[i].get_coord();
+  sum_of_local_rows[row] += data[i];
+}
+```
 
-#### Example 1: 16-bit elements
-      // Example of a 4 row x 4 column matrix using a 16-bit data element, in row-major layout.
+IMPORTANT: `get_coord` is not implemented yet.
+
+==== VNNI/Packed Layout
+Intel AMX and Intel XMX compute assumes that the B tile register
+(src1) is in the VNNI format as they need 32bit of K-data in A and B
+to be contiguous in memory.
+The VNNI blocking factor is 2 in the case of 16-bit types, and it is 4
+in the case of 8-bit types. While the current implementation assumes
+that the matrix has been already packed by the user for performance
+reasons, the layout information is needed to inform the implementation
+about this transformation.  The following example illustrates how a
+matrix in `row_major` layout is transformed into the `packed` layout
+for a 16-bit type.
```

Comment:
It is fine, we can remove details like K dimension and blocking factor. 
In the 8bit example, I think you mean extending to 8x4 so we can show the 4 last rows packing resulting in 2 rows. Otherwise, for 4x8, it will be still the same: packing for 4 second columns like the first ones resulting in 1 row.